kratos/utilities/variable_utils.cpp

```diff
@@ -486,4 +487,85 @@ void VariableUtils::AuxiliaryAtomicAdd(
         rSumValue[2] += rPrivateValue[2];
 }
 
+template <>
+ModelPart::ElementsContainerType& VariableUtils::GetContainer<ModelPart::ElementsContainerType>(ModelPart& rModelPart)
+{
+    return rModelPart.Elements();
+}
+
+template <>
+ModelPart::ConditionsContainerType& VariableUtils::GetContainer<ModelPart::ConditionsContainerType>(ModelPart& rModelPart)
+{
+    return rModelPart.Conditions();
+}
+
+template <class TDataType, class TContainerType, class TWeightDataType>
+void VariableUtils::WeightedAccumulateVariableOnNodes(
+    ModelPart& rModelPart,
+    const Variable<TDataType>& rVariable,
+    const Variable<TWeightDataType>& rWeightVariable,
+    const bool IsInverseWeightProvided)
+{
+    KRATOS_TRY
+
+    SetNonHistoricalVariableToZero(rVariable, rModelPart.Nodes());
+
+    auto& r_entities = GetContainer<TContainerType>(rModelPart);
+    const int n_entities = r_entities.size();
+
+    const std::function<double(const Node<3>&)>& r_weight_method =
+        (IsInverseWeightProvided) ?
+        static_cast<std::function<double(const Node<3>&)>>([rWeightVariable](const Node<3>& rNode) -> double {return 1.0 / rNode.GetValue(rWeightVariable);}) :
+        static_cast<std::function<double(const Node<3>&)>>([rWeightVariable](const Node<3>& rNode) -> double {return rNode.GetValue(rWeightVariable);});
```

Comment:
1. I think using std function has some overhead, use with caution in performance critical parts
2. There is no check for `rNode.GetValue(rWeightVariable)`. If it is 0 then it will crash. This happens if this var was not allocate

Reply:
1. I added the check in full debug only. Should I change it to release as well? I didnt do it considering the performance impact. (https://github.com/KratosMultiphysics/Kratos/blob/323771cd23f812b63e5dd4de498d8dadde4a3820/kratos/utilities/variable_utils.cpp#L532-L535)
2. As for the zeros, I can add a check in full debug again. Would it be sufficient?